### PR TITLE
convert into join query in getting data

### PR DESCRIPTION
**Pull request by harshtatva2002**

### CODE REVIEW

File: CI-API.Data/Repository/AdminRepository.cs [hunk 1]
```diff
@@ -855,33 +855,52 @@ public async Task<JsonResult> GetAllMissionApplication(string? search)
...
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
 
-                    MissionApplicationViewModel allMissionApplications = new()
-                    {
-                        Users = userData,
-                        Applications = missionApplications,
-                        Missions = missionData,
-                    };
+                    var allMissionApplication = await Task.FromResult(cIDbContext.MissionApplications.ToList());
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs]: Remove this line and get allMissionApplication data inside join query below, don't use separate query for that.

File: CI-API.Data/Repository/AdminRepository.cs [hunk 2]
```diff
@@ -855,33 +855,52 @@ public async Task<JsonResult> GetAllMissionApplication(string? search)
...
+
+                    return new JsonResult(new apiResponse<List<VolunteerMissionViewModel>> {  StatusCode = responseStatusCode.Success, Data = missionApplications.ToList(), Result = true });
+                    
                 }
                 else
                 {
-                    List<MissionApplication> missionApplications = cIDbContext.MissionApplications.Where(MA => MA.ApprovalStatus == StaticCode.missionApplicationPending && MA.Mission.Title.Contains(search) || MA.User.FirstName.Contains(search) || MA.User.LastName.Contains(search)).ToList();
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
+                    var allMissionApplication = await Task.FromResult(cIDbContext.MissionApplications.Where(MA => MA.ApprovalStatus == StaticCode.missionApplicationPending && MA.Mission.Title.Contains(search) || MA.User.FirstName.Contains(search) || MA.User.LastName.Contains(search)).ToList());
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs]: Remove this query and get this data inside join query.

File: CI-API.Data/Repository/AdminRepository.cs [hunk 3]
```diff
@@ -949,33 +968,70 @@ public async Task<JsonResult> GetAllStories(string? search)
...
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
 
-                    AdminPanelStoryViewModel allStoriesData = new()
-                    {
-                        Users = userData,
-                        Stories = AllStories,
-                        Missions = missionData,
-                    };
+                    var allStoriesData = await Task.FromResult(cIDbContext.Stories.Where(s => s.Status == StaticCode.storyStatusPending).ToList());
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs]: Remove this query and get this data inside join query.

File: CI-API.Data/Repository/AdminRepository.cs [hunk 4]
```diff
@@ -949,33 +968,70 @@ public async Task<JsonResult> GetAllStories(string? search)
...
+                    //};
+
+                    //return new JsonResult(new apiResponse<AdminPanelStoryViewModel> { StatusCode = responseStatusCode.Success, Data = allStoriesData, Result = false });
                 }
                 else
                 {
-                    List<Story> AllStories = cIDbContext.Stories.Where(s => s.Status == StaticCode.storyStatusPending && s.Title.Contains(search) || s.Mission.Title.Contains(search) || s.User.FirstName.Contains(search) || s.User.LastName.Contains(search)).ToList();
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
+                    var allStoriesData = await Task.FromResult(cIDbContext.Stories.Where(s => s.Status == StaticCode.storyStatusPending && s.Title.Contains(search) || s.Mission.Title.Contains(search) || s.User.FirstName.Contains(search) || s.User.LastName.Contains(search)).ToList());
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs]: Remove this query and get this data inside join query.

**Review by vimal-bhalodia-tatvasoft — APPROVED** [review]

File: CI-API.Data/Repository/AdminRepository.cs [hunk 5]
```diff
@@ -855,33 +855,47 @@ public async Task<JsonResult> GetAllMissionApplication(string? search)
 
                 if (search == null)
                 {
-                    List<MissionApplication> missionApplications = cIDbContext.MissionApplications.Where(M => M.ApprovalStatus == StaticCode.missionApplicationPending).ToList();
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
+                    var missionApplications = await Task.FromResult(from missionApplication in cIDbContext.MissionApplications.ToList()
+                                                                    join mission in cIDbContext.Missions on missionApplication.MissionId equals mission.MissionId
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs:859]: No need to add ToList() here

File: CI-API.Data/Repository/AdminRepository.cs [hunk 6]
```diff
@@ -855,33 +855,47 @@ public async Task<JsonResult> GetAllMissionApplication(string? search)
...
+                    return new JsonResult(new apiResponse<List<VolunteerMissionViewModel>> {  StatusCode = responseStatusCode.Success, Data = missionApplications.ToList(), Result = true });
+                    
                 }
                 else
                 {
-                    List<MissionApplication> missionApplications = cIDbContext.MissionApplications.Where(MA => MA.ApprovalStatus == StaticCode.missionApplicationPending && MA.Mission.Title.Contains(search) || MA.User.FirstName.Contains(search) || MA.User.LastName.Contains(search)).ToList();
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
+                    var missionApplications = await Task.FromResult(from missionApplication in cIDbContext.MissionApplications.Where(MA => MA.ApprovalStatus == StaticCode.missionApplicationPending && MA.Mission.Title.Contains(search) || MA.User.FirstName.Contains(search) || MA.User.LastName.Contains(search)).ToList()
+                                                                    join mission in cIDbContext.Missions on missionApplication.MissionId equals mission.MissionId
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs:881]: Remove .ToList()

File: CI-API.Data/Repository/AdminRepository.cs [hunk 7]
```diff
@@ -949,33 +963,43 @@ public async Task<JsonResult> GetAllStories(string? search)
 
                 if (search == null)
                 {
-                    List<Story> AllStories = cIDbContext.Stories.Where(s => s.Status == StaticCode.storyStatusPending).ToList();
-                    List<User> userData = cIDbContext.Users.ToList();
-                    List<Mission> missionData = cIDbContext.Missions.ToList();
+                    var missionStories = await Task.FromResult(from story in cIDbContext.Stories.Where(s => s.Status == StaticCode.storyStatusPending).ToList()
+                                                               join mission in cIDbContext.Missions on story.MissionId equals mission.MissionId
```
Comment (UrvishaNayee) [inline CI-API.Data/Repository/AdminRepository.cs:967]: Remove ToList()

File: CI-API.Data/Repository/AdminRepository.cs (hunk 6 above)
Comment (harshtatva2002) [reply CI-API.Data/Repository/AdminRepository.cs:881]: changes done ToList() removed

**Review by UrvishaNayee — APPROVED** [review]

**Pull request merged**